magento/magento2
# [GraphQl] Adding Wishlist GraphQl coverage

### Description (*)
This PR implements the Wishlist GraphQl

[Solution Architecture](https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/Wishlist.graphqls)

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/247


### Fixed Issues (if relevant)

1. [Customer :: MyAccount :: Wishlist Management](https://github.com/magento/community-features/issues/258)

### Manual testing scenarios (*)

Please check the following [document](https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md#options), in order to add different types of products to wishlist.

#### Add products to wishlist

```text
mutation {
  addProductsToWishlist(
    wishlistId: 0, // It may be always 0 for CE version
    wishlistItems: [
      {
        sku: "24-MB01"
        quantity: 3
      }
    ]
) {
    userInputErrors {
      code
      message
    }
    wishlist {
      id
      name
      sharing_code
      items_count
      items {
        id
        description
        qty
        added_at
      }
    }
  }
}
```

#### Update wishlist item
```text
mutation {
  updateProductsInWishlist(
    wishlistId: <wishlist_id>,
    wishlistItems: [
    {
  	wishlist_item_id: <wishlist_item_id>,
  	quantity: <quantity>,
        description: "Test Description"
    }
    ]
) {
    userInputErrors {
      code
      message
    }
    wishlist {
      id
      sharing_code
      items_count
      items {
        id
        description
        qty
        added_at
      }
    }
  }
}
```

#### Remove wishlist item
```text
mutation {
  removeProductsFromWishlist(
    wishlistId: <wishlist_id>, 
    wishlistItemsIds: [<wishlist_item_id>]
) {
    wishlist {
      name
      items_count
    }
  }
}
```

### Questions or comments


### Contribution checklist (*)
